crates/build/src/docker.rs

```diff
@@ -0,0 +1,607 @@
+            Err(_) => {
+                // sometimes we cannot remove the file due to privileged access
+                let _ = std::fs::remove_file(build_result_path);
+                anyhow::bail!(
+                    "Error parsing output from docker build. The build probably failed!"
+                )
+            }
+        };
+
+    let new_path = host_folder.join(
+        build_result
+            .target_directory
+            .as_path()
+            .strip_prefix(MOUNT_DIR)?,
+    );
+    build_result.target_directory = new_path;
+
+    let new_path = build_result.dest_wasm.as_ref().map(|p| {
+        host_folder.join(
+            p.as_path()
+                .strip_prefix(MOUNT_DIR)
+                .expect("cannot strip prefix"),
+        )
+    });
+    build_result.dest_wasm = new_path;
+
+    build_result.metadata_result.as_mut().map(|mut m| {
+        m.dest_bundle = host_folder.join(
+            m.dest_bundle
+                .as_path()
+                .strip_prefix(MOUNT_DIR)
+                .expect("cannot strip prefix"),
+        );
+        m.dest_metadata = host_folder.join(
+            m.dest_metadata
+                .as_path()
+                .strip_prefix(MOUNT_DIR)
+                .expect("cannot strip prefix"),
+        );
+        m
+    });
+    Ok(build_result)
+}
+
+/// Overwrites `build_result` and `image` fields in the metadata
+fn update_metadata(
+    build_result: &BuildResult,
+    verbosity: &Verbosity,
+    build_steps: &mut BuildSteps,
+    build_image: &ImageSummary,
+) -> Result<()> {
+    if let Some(metadata_artifacts) = &build_result.metadata_result {
+        let mut metadata = ContractMetadata::load(&metadata_artifacts.dest_bundle)?;
+        // find alternative unique identifier of the image, otherwise grab the digest
+        let image_tag = match build_image
+            .repo_tags
+            .iter()
+            .find(|t| !t.ends_with("latest"))
+        {
+            Some(tag) => tag.to_owned(),
+            None => build_image.id.clone(),
+        };
+
+        metadata.image = Some(image_tag);
+
+        crate::metadata::write_metadata(
+            metadata_artifacts,
+            metadata,
+            build_steps,
+            verbosity,
+        )?;
+    }
+    Ok(())
+}
+
+/// Creates the container and executed the build inside it
+async fn run_build(
+    build_args: String,
+    build_image: &ImageSummary,
+    build_result_path: &PathBuf,
+    contract_name: &str,
+    host_folder: &Path,
+    verbosity: &Verbosity,
+    build_steps: &mut BuildSteps,
+) -> Result<()> {
+    let client = Docker::connect_with_socket_defaults()?;
+
+    let mut entrypoint = vec!["/bin/bash".to_string(), "-c".to_string()];
+    let b_path_str = build_result_path
+        .as_os_str()
+        .to_str()
+        .context("Cannot convert Os String to String")?;
+    let mut cmds = vec![format!(
+        "mkdir -p target && cargo contract build {} --output-json > {}",
```

Comment:
Why do all other arguments get added as individual array entries and this one is its own long string? I guess it would be ugly with `vec!["cargo".to_string(), "contract".to_string()]` (like you already do above with `vec!["/bin/bash".to_string(), "-c".to_string()];`)

Reply:
The reason is because `>` and `&&` can't be represented as individual elements in the vec, so that's why I need to concat them together